server/src/test/java/org/apache/calcite/test/ServerTest.java

```diff
@@ -532,4 +533,26 @@ static Connection connect() throws SQLException {
       }
     }
   }
+
+  @Test public void testDropWithFullyQualifiedNameWhenSchemaDoesntExist() throws Exception {
+    try (Connection c = connect();
+         Statement s = c.createStatement()) {
+      checkDropWithFullyQualifiedNameWhenSchemaDoesntExist(s, "schema", "Schema");
+      checkDropWithFullyQualifiedNameWhenSchemaDoesntExist(s, "table", "Table");
+      checkDropWithFullyQualifiedNameWhenSchemaDoesntExist(s, "materialized view", "Table");
+      checkDropWithFullyQualifiedNameWhenSchemaDoesntExist(s, "view", "View");
+      checkDropWithFullyQualifiedNameWhenSchemaDoesntExist(s, "type", "Type");
+      checkDropWithFullyQualifiedNameWhenSchemaDoesntExist(s, "function", "Function");
+    }
+  }
+
+  private void checkDropWithFullyQualifiedNameWhenSchemaDoesntExist(
+      Statement statement, String objectType, String objectTypeInErrorMessage) throws Exception {
+    SQLException e = assertThrows(SQLException.class, () ->
+        statement.execute("drop " + objectType + " s.o"),
+        "expected error because the object doesn't exist");
```

Comment:
```suggestion
        "expected error from 'drop .. s.o' because schema s doesn't exist");
```

Reply:
Formally, the error is caused by the non-existent object, not the non-existent schema, this is why we check for the specific object type in the error message. 
We may reconsider what is the real cause - either a non-existing target object or a non-existing parent object - but this is not a concert of this PR.

Reply:
If you mean `because object s.o` does not exist, then please ajdust the method name accordingly

Reply:
Method name already reflects the purpose - it checks behaviour of the DROP command when it is executed with the qualified object name and the parent container doesn’t exist.

Reply:
>the parent container doesn’t exist.

That is confusing. Test name says "schema does not exist", and assertion `reason` says "object does not exist".

Did you intend to add tests for the case when schema exists, yet the object is missing?

Reply:
same question，+1

Reply:
The description of the test case is not necessarily the same as the error message. The test case checks the behavior of the `DROP` command when the schema doesn't exist. If the schema doesn't exist, any object inside the schema doesn't exist either. Currently, Apache Calcite throws errors like `table not found` for such cases. Therefore, the current assertion message is correct - it expresses the same problem as we show to the user.

As I mentioned before, perhaps this is not the correct semantics. For example, Postgres will throw `schema not found`, which I think is a better error description. But this is not the concern of the given PR.